zebra-chain/src/transaction.rs

```diff
@@ -130,4 +130,33 @@ impl Transaction {
             Transaction::V4 { expiry_height, .. } => Some(*expiry_height),
         }
     }
+    /// Get if this transaction is coinbase.
+    pub fn is_coinbase(&self) -> Option<bool> {
+        match self {
+            Transaction::V1 { ref inputs, .. } => {
+                inputs.iter().next().and_then(|input| match input {
+                    TransparentInput::Coinbase { .. } => Some(true),
+                    _ => None,
+                })
+            }
+            Transaction::V2 { ref inputs, .. } => {
+                inputs.iter().next().and_then(|input| match input {
+                    TransparentInput::Coinbase { .. } => Some(true),
+                    _ => None,
+                })
+            }
+            Transaction::V3 { ref inputs, .. } => {
+                inputs.iter().next().and_then(|input| match input {
+                    TransparentInput::Coinbase { .. } => Some(true),
+                    _ => None,
+                })
+            }
+            Transaction::V4 { ref inputs, .. } => {
+                inputs.iter().next().and_then(|input| match input {
+                    TransparentInput::Coinbase { .. } => Some(true),
+                    _ => None,
+                })
+            }
+        }
```

Comment:
Suggestion: I think the coinbase check can be simplified to something like:
```
       self.inputs().next().and_then(|input| match input {
                TransparentInput::Coinbase { .. } => true,
                _ => false,
            })
```

Issue: We should check all the inputs, not just the first one:
```
    /// Does this transaction contain any coinbase inputs?
    pub fn contains_coinbase_input(&self) -> bool {
        self.inputs().any(|input| match input {
                TransparentInput::Coinbase { .. } => true,
                _ => false,
            })
    }
```

Reply:
Makes sense. I replaced the `is_coinbase` with suggested `contains_coinbase_input`.